src/client/extensionActivation.ts

```diff
@@ -71,7 +73,11 @@ async function activateLegacy(
         (context.extensionMode === ExtensionMode.Development ||
             workspace.getConfiguration('jupyter').get<boolean>('development', false));
     serviceManager.addSingletonInstance<boolean>(IsDevMode, isDevMode);
-    const isPreRelease = isDevMode || context.extension.packageJSON?.__metadata?.preRelease;
+    const packageJSONLive = JSON.parse(
+        fsExtra.readFileSync(path.join(context.extensionPath, 'package.json'), { encoding: 'utf-8' })
```

Comment:
Can we make this async, this actually slows loading this extension and slows other extensions with this sync operation.

Reply:
Then it would have to be a separate service because this flag needs to be set before other items in the container are created.

I guess it could be an async function? Instead of setting a boolean, set a promise?

Reply:
> set a promise

Yes, we can just give the promise i.e. turn isPreRelease into a `Promise<boolean>`